server/lib/routes/redirect-download-firefox.js

```diff
@@ -0,0 +1,79 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+'use strict';
+
+/**
+ * The sole purpose of this route is to log clicks on the
+ * "Download latest Firefox" button on the /update_firefox page,
+ * allowing that page to use no JS.
+ */
+
+const amplitude = require('../amplitude');
+const { logFlowEvent } = require('../flow-event');
+
+const validation = require('../validation');
+
+const {
+  ACTION: ACTION_TYPE,
+  FLOW_ID: FLOW_ID_TYPE,
+  HEX32: HEX32_TYPE,
+  OFFSET: OFFSET_TYPE,
+  STRING: STRING_TYPE,
+  UTM: UTM_TYPE,
+  UTM_CAMPAIGN: UTM_CAMPAIGN_TYPE,
+} = validation.TYPES;
+
+const {
+  CONTEXT: CONTEXT_PATTERN,
+  ENTRYPOINT: ENTRYPOINT_PATTERN,
+  SERVICE: SERVICE_PATTERN,
+} = validation.PATTERNS;
+
+module.exports = function (config) {
+  const DOWNLOAD_FIREFOX_FLOW_EVENT_NAME = 'flow.update-firefox.link.download';
+
+  return {
+    method: 'get',
+    path: '/download_firefox',
+    validate: {
+      query: {
+        action: ACTION_TYPE.optional(),
+        context: STRING_TYPE.regex(CONTEXT_PATTERN).required(),
+        deviceId: HEX32_TYPE.required(),
+        entrypoint: STRING_TYPE.regex(ENTRYPOINT_PATTERN).optional(),
+        flowBeginTime: OFFSET_TYPE.required(),
+        flowId: FLOW_ID_TYPE.required(),
+        service: STRING_TYPE.regex(SERVICE_PATTERN).optional(),
+        'utm_campaign': UTM_CAMPAIGN_TYPE.optional(),
+        'utm_content': UTM_TYPE.optional(),
+        'utm_medium': UTM_TYPE.optional(),
+        'utm_source': UTM_TYPE.optional(),
+        'utm_term': UTM_TYPE.optional()
+      }
+    },
+    process: function (req, res) {
+      const metricsData = req.query;
+
+      const flowBeginTime = parseInt(metricsData.flowBeginTime);
+      metricsData.flowBeginTime = flowBeginTime;
+
+      const time = Date.now();
+
+      amplitude({
+        flowTime: flowBeginTime,
+        time,
+        type: DOWNLOAD_FIREFOX_FLOW_EVENT_NAME
```

Comment:
@philbooth - I imagine I'll need to update something in the [FUZZY_EVENTS](https://github.com/shane-tomlinson/fxa-content-server/blob/7e496bf182843bc1ce6b6317957d5e88adddd08c/server/lib/amplitude.js#L55) for this to be logged, right? None of the [existing groups in fxa-shared's amplitude module](https://github.com/mozilla/fxa-shared/blob/master/metrics/amplitude.js#L17) make much sense.

Am I making this too difficult? Do we need to track clicks on these?

What I'm hoping to avoid is loading the whole JS bundle into the `/update_firefox` route because that route is supposed to serve old browsers, and if we are using JS that doesn't work on those old browsers then the user can't really do anything.

Reply:
> I'll need to update something in the [FUZZY_EVENTS](https://github.com/shane-tomlinson/fxa-content-server/blob/7e496bf182843bc1ce6b6317957d5e88adddd08c/server/lib/amplitude.js#L55) for this to be logged, right?

Yep.

> None of the [existing groups in fxa-shared's amplitude module](https://github.com/mozilla/fxa-shared/blob/master/metrics/amplitude.js#L17) make much sense.

@irrationalagent and @davismtl are in charge of the event taxonomy, worth looping them in for guidance on the appropriate group etc.

> Am I making this too difficult?

Having given it some thought, the direction you've taken here is exactly correct I think.

> Do we need to track clicks on these?

Makes sense to do so I reckon.